altair/vegalite/v5/api.py

```diff
@@ -3865,101 +3850,81 @@ def interactive(  # noqa: C901
+                nonreactive_domain = param(
+                    react=False, expr=initial_computed_domain.name
                 )
-
-            if legend_encoding is not None:
-                if utils.is_undefined(
-                    interactive_chart.encoding[legend_encoding]["type"]
-                ):
-                    legend_encoding_type = interactive_chart.encoding[
-                        legend_encoding
-                    ].to_dict(context={"data": interactive_chart.data})["type"]
+                scale = facet_legend["scale"]
+                if utils.is_undefined(scale):
+                    scale = {"domain": nonreactive_domain}
                 else:
-                    legend_encoding_type = interactive_chart.encoding[legend_encoding][
-                        "type"
-                    ]
-                if (
-                    legend_encoding_type == "nominal"
-                ):  # TODO Ideally this would work for ordinal data too
-                    legend_selection = selection_point(
-                        bind="legend", encodings=[legend_encoding]
-                    )
-                    initial_computed_domain = param(expr=f"domain('{legend_encoding}')")
-                    nonreactive_domain = param(
-                        react=False, expr=initial_computed_domain.name
-                    )
-                    if utils.is_undefined(
-                        interactive_chart.encoding[legend_encoding]["scale"]
-                    ):
-                        interactive_chart.encoding[legend_encoding]["scale"] = {
-                            "domain": nonreactive_domain
-                        }
-                    else:
-                        interactive_chart.encoding[legend_encoding]["scale"][
-                            "domain"
-                        ] = nonreactive_domain
-
-                    interactive_chart = interactive_chart.add_params(
-                        legend_selection, initial_computed_domain, nonreactive_domain
-                    ).transform_filter(legend_selection)
-        return interactive_chart
+                    scale["domain"] = nonreactive_domain
+                chart = chart.add_params(
+                    legend_selection,
+                    initial_computed_domain,
+                    nonreactive_domain,
+                ).transform_filter(legend_selection)
+            else:
+                msg = f"Expected only 'nominal' legend type but got {legend_type!r}"
+                raise NotImplementedError(msg)
+        return chart
+
+
+LegendChannel_T: TypeAlias = Literal[
+    "color",
+    "fill",
+    "shape",
+    "stroke",
+    "opacity",
+    "fillOpacity",
+    "strokeOpacity",
+    "strokeWidth",
+    "strokeDash",
+    "angle",  # TODO Untested
+    "radius",  # TODO Untested
+    "radius2",  # TODO Untested
+    # "size",  # TODO Currently size is not working, renders empty legend
+]
+
+
+def _add_tooltip(chart: _TChart, /) -> _TChart:
```

Comment:
Mainly thinking ahead here for all the other `ChartType`s.

I imagine this would be preferred to simply repeating the code in each method.